rust/crd/src/authentication.rs

```diff
@@ -33,67 +42,137 @@ pub enum Error {
-                    ),
-                })?;
-
-        let resolved_auth_class_name = resolved_auth_class.name_any();
-
-        match &resolved_auth_class.spec.provider {
-            AuthenticationClassProvider::Static(_) => {}
-            AuthenticationClassProvider::Ldap(ldap) => {
-                if ldap.tls.uses_tls() && !ldap.tls.uses_tls_verification() {
-                    NoLdapTlsVerificationNotSupportedSnafu {
-                        authentication_class: ObjectRef::<AuthenticationClass>::new(
-                            &resolved_auth_class_name,
-                        ),
+                .context(AuthenticationClassRetrievalFailedSnafu)?;
+
+            let auth_class_name = auth_class.name_any();
+
+            match &auth_class.spec.provider {
+                AuthenticationClassProvider::Static(provider) => {
+                    resolved_auth_classes.push(AuthenticationClassResolved::Static {
+                        provider: provider.to_owned(),
+                    })
+                }
+                AuthenticationClassProvider::Ldap(provider) => {
+                    if provider.tls.uses_tls() && !provider.tls.uses_tls_verification() {
+                        NoLdapTlsVerificationNotSupportedSnafu {
+                            authentication_class: ObjectRef::<AuthenticationClass>::new(
+                                &auth_class_name,
+                            ),
+                        }
+                        .fail()?
                     }
-                    .fail()?
+                    resolved_auth_classes.push(AuthenticationClassResolved::Ldap {
+                        provider: provider.to_owned(),
+                    })
                 }
+                AuthenticationClassProvider::Oidc(provider) => {
+                    resolved_auth_classes.push(AuthenticationClassResolved::from_oidc(
+                        &auth_class_name,
+                        provider,
+                        entry,
+                        nifi.clone(),
+                    )?)
+                }
+                _ => AuthenticationClassProviderNotSupportedSnafu {
+                    authentication_class_provider: auth_class.spec.provider.to_string(),
+                    authentication_class: ObjectRef::<AuthenticationClass>::new(&auth_class_name),
+                }
+                .fail()?,
+            };
+        }
+
+        Ok(resolved_auth_classes)
+    }
+
+    fn from_oidc(
+        auth_class_name: &str,
+        provider: &oidc::AuthenticationProvider,
+        auth_details: &ClientAuthenticationDetails,
+        nifi: NifiCluster,
+    ) -> Result<AuthenticationClassResolved> {
+        let oidc_provider = match &provider.provider_hint {
+            None => {
+                info!("No OIDC provider hint given in AuthClass {auth_class_name}, assuming {default_oidc_provider_name}",
+                    default_oidc_provider_name = serde_json::to_string(&DEFAULT_OIDC_PROVIDER).unwrap());
```

Comment:
I *think* we can use https://docs.rs/strum/latest/strum/ for that, which is what we do in other places. Did not test it though.

But this requires changes in op-rs, so I'd consider this optional

Reply:
After removing provider hint this is also not relevant anymore (at least here).